server/api.go

```diff
@@ -1,83 +1,64 @@
 package server
 
 import (
-	"context"
 	"net/http"
 
 	"github.com/Hukyl/genesis-kma-school-entry/models"
-	"github.com/Hukyl/genesis-kma-school-entry/settings"
 	"github.com/gin-gonic/gin"
 )
 
-// wrapWithContext is a helper function that wraps a handler function
-// with a context and returns a gin.HandlerFunc.
-func wrapWithContext(ctx context.Context, fn func(context.Context, *gin.Context)) gin.HandlerFunc {
-	return func(c *gin.Context) {
-		fn(ctx, c)
-	}
+type UserRepository interface {
+	Exists(user *models.User) (bool, error)
+	Create(user *models.User) error
 }
 
-// GetRate is a handler that fetches the exchange rate between USD and UAH
-// from the National Bank of Ukraine API and returns it as a JSON response.
-func GetRate(ctx context.Context, c *gin.Context) {
-	apiClient, ok := ctx.Value(settings.APIClientKey).(Client)
-	if !ok {
-		c.JSON(http.StatusInternalServerError, "")
-		return
-	}
-	rateFetcher := apiClient.RateFetcher
-	rate, err := rateFetcher.FetchRate("USD", "UAH")
-	if err != nil {
-		c.JSON(http.StatusBadRequest, err.Error())
-		return
+// GetRateWrapper is a handler that fetches the exchange rate between USD and UAH
+// from a RateFetcher interface and returns it as a JSON response.
+func GetRateWrapper(rateFetcher RateFetcher) func(*gin.Context) {
+	return func(c *gin.Context) {
+		rate, err := rateFetcher.FetchRate("USD", "UAH")
```

Comment:
OPTIONAL: It's better to move currency codes to constants.
```go
const (
    USD = "USD"
    UAH = "UAH" 
)
```

Reply:
while it's used only once in the code, imo it's okay to leave it like this